### DISCUSSION TITLE
Add lyrics / karaoke mode

### DISCUSSION THREAD

**Discussion by Aaantoni** [Selbi182/SpotifyBigPicture#24] [Ideas]
Love the app (and the discovery one too!)

Would there be a possibility to add either the built-in synced lyrics from Spotify (or more specifically, Musixmatch since that's the service they use) or un-synced but more accurate lyrics from Genius? Ideally, they would either occupy the space currently reserved for album/playlist tracklists, or have their own "karaoke" preset. I would presume adding Genius lyrics would be more difficult, since it's not officially supported by Spotify, but it's already been done by other third-party extensions, such as **[Spicetify](https://github.com/spicetify/spicetify-cli/blob/master/CustomApps/lyrics-plus/ProviderGenius.js)**, and would honestly be my preferred option.

**Comment by Selbi182** [comment]
Thank you!

I thought about this myself a couple of times, but I got a lot of design issues to figure out first. The biggest problem is that Spotify's integration for the lyrics is not publicly accessible. So that means I gotta resort to Musixmatch, but that one has the other issue with its price model: 
![grafik](https://github.com/Selbi182/SpotifyBigPicture/assets/8850085/809caa64-7d4b-4f25-a406-a93e7308b78a)
Meaning I cannot just directly build it into the app, as 2k requests per day is not a lot at all. I'd have the user of SpotifyBigPicture create their own API access. Possible, but annoying.

Then we get another problem: The visual design. There are so many questions to ask. How would this mode be activated? From a button? Automatically? How would this be activated with read-only displays, such as being hosted on a Raspberry Pi with no immediate access? Remote control? How should it be displayed? An full overlay is simple but would hide everything else. Replacing the tracklist works, but there won't be much space available, regardless of whether it's synchronized or not.

And lastly: Synchronization wouldn't really work, because the interface is CONSTANTLY out of sync with the internal API by a few seconds.

You see there are a lot of questions to be tackled first. If you'd like to help me brainstorm the design, you could add me on Discord (`selbi`) and we can figure something out :)
